kong/plugins/session/schema.lua

```diff
@@ -92,6 +92,7 @@ return {
           { remember_absolute_timeout = { type = "number", default = 2592000 } },
           { response_headers = headers },
           { request_headers = headers },
+          { read_bodies = { type = "boolean", default = false } },
```

Comment:
```suggestion
          { read_request_body_on_logout = { type = "boolean", default = false } },
```

nit: if we plan on expanding the usage on flows other than logout, then we can drop `_on_logout`. Not a blocker for merging.

Reply:
The `logout` is kinda magical as it is always enabled (so this plugin always tries to check if the request is logout based on config parameters). Perhaps we should add option like `disable_logout=true`, would that make more sense? Ping @hbagdi (@hanshuebner what's your take?)

Reply:
> would that make more sense?

How would that fix the problem of reading bodies by default though?

Reply:
@hbagdi because it only reads bodies when it detects logout. If it is not detecting logout it does not.

Reply:
just to make sure I understood, @bungle, you mean we could add a new `enable_logout` option to replace this `read_bodies`, but serving the same purpose as `read_bodies` in the current implementation, i.e.  if `enable_logout` then read `logout_post_arg`? Also, this new option needs a new entry in [removed_fields](https://github.com/Kong/kong/blob/master/kong/clustering/compat/removed_fields.lua) right?

Reply:
@samugi:

The options we have:
1. remove default (tried, but @hbagdi wanted new field)
2. `read_bodies` added, but was thinking, should we have `enable_logout` instead, which is disabled by default, but when enabled, it would read bodies.
4. setting actively `logout_post_arg` to `null`. This works already, but `null` is considered problematic with Yaml, CRDs, etc. (the 1. was removing default so it would be null/nil by default).

Reply:
@samugi I meant that `enable_logout` will disable logout totally by default, and if you need plugin to work for logout you do `enable_logout=true`. It means that we skip the whole logout handling with this.

Reply:
`enable_logout` would enable this block:
https://github.com/Kong/kong/blob/master/kong/plugins/session/access.lua#L80-L93

By default it is never executed then.